sink_service/source/main.c

```diff
@@ -60,17 +60,79 @@ static bool get_service_name(char service_name[MAX_SIZE_SERVICE_NAME], unsigned
     return true;
 }
 
+/**
+ * \brief   Attempts to acquire an environment variable
+ * \param   dest
+ *          The destination buffer where to store the read value
+ * \param   name
+ *          The name of the environment variable to acquire
+ * \return  A pointer to the character buffer
+ */
+static char * get_env_var(char * name)
+{
+    char * ptr = NULL;
+    if ((ptr = getenv(name)) != NULL)
+    {
+        LOGI("%s: %lu\n", name, ptr);
+    }
+    return ptr
+}
+
+/**
+ * \brief   Obtains environment parameters to control sink settings
+ * \param   bitrate
+ *          Pointer where to store bitrate value (if any)
+ * \param   port_name
+ *          Pointer where to store port_name value (if any)
+ * \param   sink_id
+ *          Pointer where to store sink_id value (if any)
+ * \param   max_poll_fail_duration
+ *          Pointer where to store max_poll_fail_duration value (if any)
+ */
+static void get_env_parameters(unsigned long * bitrate,
+                               char * port_name,
+                               unsigned int * sink_id,
+                               unsigned int * max_poll_fail_duration)
+{
+    char * ptr;
+
+    if ((ptr = getenv("WM_GW_SINK_BITRATE")) != NULL)
+    {
+        *bitrate = strtoul(ptr, NULL, 0);
+        LOGI("WM_GW_SINK_BITRATE: %lu\n", *bitrate);
+    }
+    if ((ptr = getenv("WM_GW_SINK_ID")) != NULL)
+    {
+        *sink_id = strtoul(ptr, NULL, 0);
+        LOGI("WM_GW_SINK_ID: %lu\n", *sink_id);
+    }
+    if ((ptr = getenv("WM_GW_SINK_UART_PORT")) != NULL)
+    {
+        port_name = ptr;
```

Comment:
an *port_name_p = ptr here

Reply:
Thanks, rookie mistake ;)

Reply:
You're totally right on this and thinking it through it is clear that it should not "work" otherwise. I was lucky that I did not get a core dump. Also I noticed my mistake was with the print. I forgot to update the variable to print the pointer, thus I got the garbled output.